qutip/solver/result.py

```diff
@@ -895,7 +970,31 @@ def __repr__(self):
         lines.append(">")
         return "\n".join(lines)
 
-    def __add__(self, other):
+    def merge(self, other, p=None):
```

Comment:
The `__add__` is already in the users hand and part of the official interface.
It is documented in the [guide](https://qutip.readthedocs.io/en/qutip-5.0.x/guide/dynamics/dynamics-monte.html#changing-the-number-of-trajectories) (and maybe the tutorials).
It's new, so it's probably not used much, but still we shouldn't remove it without a deprecation warning or a major release...

It would be better to keep it as merging with the default value of `p`.